### PR TITLE
feat(withdrawal): hardfork needed distinguish withdrawal types

### LINKED ISSUE #395: Use Validator field in EngineAPI withdrawals to distinguish unstake and reward distribution
## Description and context
Currently we have two types of withdrawals. One is unstaking that is triggered by users. The other is reward distribution which happens automatically. From the outside, there is no way to know which type of withdrawal it is. We should find a way to distinguish them. This will help indexers and other apps to show correct information to the users.

## Suggested solution
Since the Validator field is currently set to 0 and not used by EL, we can use this field to mark the type of withdrawals. 0 for unstaking, 1 for rewards.

### PR DESCRIPTION
**Pull request by ezreal1997**
Use `Validator` field in EngineAPI withdrawals to distinguish unstake, reward and ubi distribution

issue: closes #395

### CODE REVIEW

**Review by 0xHansLee — COMMENTED** [review]
This might require hardfork, right? Then, let's edit the title of PR to notice that upgrade handler is required for this.

**Review by 0xHansLee — APPROVED** [review]
If the upgrade handler for this change is implemented in a separate PR, looks good to me.

**Review by jdubpark — APPROVED** [review]

**Comment by ezreal1997** [comment]
> This might require hardfork, right? Then, let's edit the title of PR to notice that upgrade handler is required for this.

hardfork needed, but we don't need any data migration since proto3 is both forward and backward compatible

**Review by edisonz0718 — APPROVED** [review]

File: client/x/evmstaking/keeper/withdraw.go [hunk 1]
```diff
@@ -60,9 +60,10 @@ func (k Keeper) ProcessUnbondingWithdrawals(ctx context.Context, unbondedEntries
 			uint64(sdk.UnwrapSDKContext(ctx).BlockHeight()),
 			delEvmAddr,
 			entry.Amount.Uint64(),
+			types.WithdrawalType_WITHDRAWAL_TYPE_UNSTAKE,
 		))
 		if err != nil {
-			return errors.Wrap(err, "add unbonding withdrawal to queue")
```
Comment (edisonz0718) [inline client/x/evmstaking/keeper/withdraw.go:65]: nit: unbonded may be a better word to use here

**Pull request merged**